### ISSUE TITLE
Set Batchnorm weight scalar initialization to unit (not random uniform)

### ISSUE THREAD

**Issue by cstein06** [pytorch/pytorch#12259]
## 🐛 Feature suggestion

Pytorch Batch normalization currently initializes it's scalar weight parameter with random uniform variables (https://github.com/pytorch/pytorch/blob/1d3f650ce4f1b781f03e8b4f250a25d5a8f819cc/torch/nn/modules/batchnorm.py#L46). This initialization has no theoretical basis, while on the other hand it makes sense to use a fixed value of 1. 

Part of the usefulness of Batchnorm is that it can initialize the network easily, with inputs to hidden units normalized to zero norm and unit variance (original paper: https://arxiv.org/abs/1502.03167) and it is also how recent studies use it (e.g. https://arxiv.org/abs/1706.02677).

## To change

From 
init.uniform_(self.weight)

to 
init.ones_(self.weight)

**Comment by vishwakftw** [comment]
Won’t this cause BC problems?

**Comment by Kaixhin** [comment]
This is tricky, but arguably this is a bug and should be fixed, as the layer originally came with a well-specified initialisation and this was not put in properly. I can't say for certain, but this may even be a cause for some reported discrepancies in performance between PyTorch and other frameworks.

**Comment by yaysummeriscoming** [comment]
+1, I came across this independently myself just now when debugging one of my models.  I did find a small performance drop using random initialization compared to unit initialization.  Unit initialization is as per the paper?
